### PR TITLE
CAV-327 - Update hmrc libraries to the latest version

### PR DESCRIPTION
**Pull request by brdgeth**
Update HMRC libraries.
Fixed unit tests:
- FakeRequest is a POST (not the default of GET)
- Location header url does not use routes anymore and as a result the unit tests produce the same results and pass in both IntelliJ and the command line

### CODE REVIEW

**Review by johno1985 — APPROVED** [review]

**Comment by johno1985** [comment]
rebuild

**Comment by johno1985** [comment]
I believe if this branch is rebased with main then the tests will pass

File: test/uk/gov/hmrc/cipphonenumberfrontend/controllers/OtpControllerSpec.scala [hunk 1]
```diff
@@ -132,7 +132,7 @@ class OtpControllerSpec extends AnyWordSpec
   }
 
   trait SetUp {
-    protected val fakeRequest = FakeRequest()
+    protected val fakeRequest = FakeRequest(POST, "")
```
Comment (johno1985) [inline test/uk/gov/hmrc/cipphonenumberfrontend/controllers/OtpControllerSpec.scala:135]: I believe this makes no difference, can be left as was

File: test/uk/gov/hmrc/cipphonenumberfrontend/controllers/VerifyControllerSpec.scala [hunk 2]
```diff
@@ -121,7 +121,7 @@ class VerifyControllerSpec extends AnyWordSpec
   }
 
   trait SetUp {
-    protected val fakeRequest = FakeRequest()
+    protected val fakeRequest = FakeRequest(POST, "")
```
Comment (johno1985) [inline test/uk/gov/hmrc/cipphonenumberfrontend/controllers/VerifyControllerSpec.scala:124]: I believe this makes no difference, can be left as was

**Review by johno1985 — COMMENTED** [review]
minor comment

File: test/uk/gov/hmrc/cipphonenumberfrontend/controllers/OtpControllerSpec.scala (hunk 1 above)
Comment (brdgeth) [reply test/uk/gov/hmrc/cipphonenumberfrontend/controllers/OtpControllerSpec.scala:135]: It was defaulting to a GET and failing the unit tests

Comment (brdgeth) [reply test/uk/gov/hmrc/cipphonenumberfrontend/controllers/OtpControllerSpec.scala:135]: It went to the invalid block and had a 400.
`      invalid => {
        logger.warn(s"Failed to validate request")
        Future.successful(BadRequest(verifyPage(invalid)))
      },`

Comment (johno1985) [reply test/uk/gov/hmrc/cipphonenumberfrontend/controllers/OtpControllerSpec.scala:135]: That's odd, they were not failing before now

Comment (brdgeth) [reply test/uk/gov/hmrc/cipphonenumberfrontend/controllers/OtpControllerSpec.scala:135]: yeah its odd, they must have changed something.
the tests pass on Jenkins and the command line (but not IntelliJ).  I have a green build on here.

**Comment by brdgeth** [comment]
@johno1985 if you are happy can you resolve the 2 above and approve this one please?

File: test/uk/gov/hmrc/cipphonenumberfrontend/controllers/VerifyControllerSpec.scala (hunk 2 above)
Comment (mar36tin) [reply test/uk/gov/hmrc/cipphonenumberfrontend/controllers/VerifyControllerSpec.scala:124]: Agree

File: test/uk/gov/hmrc/cipphonenumberfrontend/controllers/OtpControllerSpec.scala (hunk 1 above)
Comment (mar36tin) [reply test/uk/gov/hmrc/cipphonenumberfrontend/controllers/OtpControllerSpec.scala:135]: okay i see if it defaults to GET then add it

Comment (brdgeth) [reply test/uk/gov/hmrc/cipphonenumberfrontend/controllers/OtpControllerSpec.scala:135]: The no-arg constructor of FakeRequest uses a default http configuration which defaults to GET.  Therefore Play has a validation error, 400.

File: test/uk/gov/hmrc/cipphonenumberfrontend/controllers/VerifyControllerSpec.scala (hunk 2 above)
Comment (brdgeth) [reply test/uk/gov/hmrc/cipphonenumberfrontend/controllers/VerifyControllerSpec.scala:124]: see above comment

File: test/uk/gov/hmrc/cipphonenumberfrontend/controllers/OtpControllerSpec.scala (hunk 1 above)
Comment (brdgeth) [reply test/uk/gov/hmrc/cipphonenumberfrontend/controllers/OtpControllerSpec.scala:135]: I tested it again today and same issue, so I will leave the POST in the code

**Comment by johno1985** [comment]
Happy to approve

**Review by johno1985 — APPROVED** [review]
Approve, since @mar36tin also made comments he may need to approve also

**Review by mar36tin — APPROVED** [review]

**Pull request merged**